nixos/modules/profiles/base.nix

```diff
@@ -49,7 +49,7 @@
   ];
 
   # Include support for various filesystems.
-  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "zfs" "ntfs" "cifs" ];
+  boot.supportedFilesystems = lib.mkDefault [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "zfs" "ntfs" "cifs" ];
```

Comment:
I think you could condition including `zfs` based on the kernel version from `config.boot.kernelPackages`

Reply:
^ seems interesting for sure. I'm a bit concerned about using mkDefault. Would users who have already added an additional FS find their system unbootable?

Reply:
Hmm. Looks like it can't be done through comparing `config.boot.kernelPackages.kernel.version`; infinite recursion, which I think is coming from the zfs task, which also looks at supportedFilesystems, and `config.boot.kernelPackages`. Though I'm unsure about the diagnosis.

Reply:
What's the trace with --show-trace?

Reply:
Here it is:

<details><summary>Trace</summary>
```
error: while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release-combined.nix:20:34, called from undefined position:
while evaluating 'removeMaintainers' at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release-combined.nix:17:23, called from /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release-combined.nix:20:37:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release.nix:146:32, called from /Users/samuel/tmp/nixpkgs/nixpkgs/lib/attrsets.nix:292:43:
while evaluating 'makeIso' at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release.nix:42:5, called from /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release.nix:146:40:
while evaluating 'hydraJob' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/customisation.nix:157:14, called from /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/release.nix:46:5:
while evaluating the attribute 'drvPath' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/customisation.nix:174:13:
while evaluating the attribute 'drvPath' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/customisation.nix:141:13:
while evaluating the attribute 'sources' of the derivation 'nixos-minimal-19.09pre56789.gfedcba-x86_64-linux.iso' at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/lib/make-iso9660-image.nix:49:3:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/lib/make-iso9660-image.nix:56:18, called from undefined position:
while evaluating the attribute 'source' at /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/modules/installer/cd-dvd/iso-image.nix:585:11:
while evaluating the attribute 'boot.kernelPackages.kernel' at undefined position:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:338:9:
while evaluating the option `boot.kernelPackages':
while evaluating the attribute 'isDefined' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:375:5:
while evaluating the attribute 'values' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:364:9:
while evaluating the attribute 'values' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:458:7:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:350:28, called from /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:350:17:
while evaluating 'dischargeProperties' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:417:25, called from /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:351:62:
while evaluating the attribute 'condition' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:502:14:
while evaluating the attribute 'condition' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:502:14:
while evaluating the attribute 'condition' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:502:14:
while evaluating the attribute 'boot.supportedFilesystems' at undefined position:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:338:9:
while evaluating the option `boot.supportedFilesystems':
while evaluating the attribute 'isDefined' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:375:5:
while evaluating the attribute 'values' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:364:9:
while evaluating the attribute 'values' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:458:7:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:350:28, called from /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:350:17:
while evaluating 'dischargeProperties' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:417:25, called from /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:351:62:
while evaluating the attribute 'value' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:234:44:
while evaluating 'optional' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/lists.nix:241:20, called from /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/modules/profiles/base.nix:64:6:
while evaluating 'versionOlder' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/strings.nix:461:22, called from /Users/samuel/tmp/nixpkgs/nixpkgs/nixos/modules/profiles/base.nix:66:5:
while evaluating the attribute 'boot.kernelPackages' at undefined position:
while evaluating anonymous function at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:338:9:
infinite recursion encountered, at /Users/samuel/tmp/nixpkgs/nixpkgs/lib/modules.nix:338:9
```
</details>